yfinance/scrapers/history.py

```diff
@@ -132,16 +132,23 @@ def history(self, period="1mo", interval="1d",
             else:
                 end = utils._parse_user_dt(end, tz)
             if start is None:
+                SECONDS_IN_A_DAY = 86400
+                SECONDS_IN_A_YEAR = 31536000
                 if interval == "1m":
-                    start = end - 604800   # 7 days
-                elif interval in ("5m", "15m", "30m", "90m"):
-                    start = end - 5184000  # 60 days
-                elif interval in ("1h", '60m'):
-                    start = end - 63072000  # 730 days
+                    start = end - (SECONDS_IN_A_DAY * 8)  # 8 days / max chunk for 1m
+                elif interval in ("2m", "5m", "15m", "30m", "90m"):
+                    start = end - (SECONDS_IN_A_DAY * 60)  # 60 days / max chunk for these
+                elif interval in ("1h", "60m"):
+                    start = end - (SECONDS_IN_A_DAY * 730)  # 730 days / max chunk for 1h
                 else:
-                    start = end - 3122064000  # 99 years
+                    start = end - (SECONDS_IN_A_YEAR * 99)  # 99 years / max chunk for day+ intervals
             else:
                 start = utils._parse_user_dt(start, tz)
+                
+            # Program processing time adds seconds to request 
+            # To avoid exceeding max period, must subtract a buffer of seconds
+            ARBITRARY_PROCESS_TIME_BUFFER = 60 # seconds
+            start += ARBITRARY_PROCESS_TIME_BUFFER # moves start forward in time
```

Comment:
I'm sensing over-engineering here. Can simplify to:

> start += 60  # allow for processing time

And I'd drop delay to 5 seconds. How slow can processing & Yahoo be?